trade-finance-manager-ui/server/controllers/utilisation-reports/update-utilisation-report-status/index.ts

```diff
@@ -0,0 +1,87 @@
+import { Request, Response } from 'express';
+import api from '../../../api';
+import { asString } from '../../../helpers/validation';
+import { ReportIdentifier, ReportWithStatus } from '../../../types/utilisation-reports';
+import { UTILISATION_REPORT_RECONCILIATION_STATUS } from '../../../constants';
+import { getUtilisationReports } from '..';
+import { TfmSessionUser } from '../../../types/tfm-session-user';
+
+const CHECKBOX_PREFIX_REGEX = 'set-status--';
+const CHECKBOX_ID_TYPE_REGEX = 'reportId|bankId';
+const MONGO_ID_REGEX = '[a-f\\d]+';
+const CHECKBOX_PATTERN = {
+  WITHOUT_GROUPS: new RegExp(CHECKBOX_PREFIX_REGEX),
+  WITH_GROUPS: new RegExp(`${CHECKBOX_PREFIX_REGEX}(?<idType>${CHECKBOX_ID_TYPE_REGEX})-(?<id>${MONGO_ID_REGEX})`),
+} as const;
+
+const FORM_BUTTON_VALUES = {
+  COMPLETED: 'completed',
+  NOT_COMPLETED: 'not-completed',
+} as const;
+
+const getReportIdentifiersFromBody = (body: unknown, month: number, year: number): ReportIdentifier[] => {
+  if (!body || typeof body !== 'object') {
+    return [];
+  }
+
+  return Object.keys(body)
+    .filter((key) => key.match(CHECKBOX_PATTERN.WITHOUT_GROUPS))
+    .map((setStatusKey): ReportIdentifier => {
+      const match = setStatusKey.match(CHECKBOX_PATTERN.WITH_GROUPS);
+      if (!match?.groups) {
+        throw new Error(`Failed to parse report identifiers from request body key '${setStatusKey}'`);
+      }
+
+      const { idType, id } = match.groups;
+      switch (idType) {
+        case 'reportId':
+          return { id };
+        case 'bankId':
+          return { bankId: id, month, year };
+        default:
+          throw new Error('Failed to parse report identifiers from request body');
+      }
+    });
+};
+
+const getReportWithStatus = (reportIdentifier: ReportIdentifier, formButton: string): ReportWithStatus | undefined => {
+  switch (formButton) {
+    case FORM_BUTTON_VALUES.COMPLETED:
+      return {
+        status: UTILISATION_REPORT_RECONCILIATION_STATUS.RECONCILIATION_COMPLETED,
+        report: reportIdentifier,
+      };
+    case FORM_BUTTON_VALUES.NOT_COMPLETED:
+      if ('bankId' in reportIdentifier) {
+        return undefined;
+      }
+      return {
+        status: UTILISATION_REPORT_RECONCILIATION_STATUS.PENDING_RECONCILIATION,
+        report: reportIdentifier,
+      };
+    default:
+      throw new Error('Button query parameter matched no known values');
```

Comment:
Should add `formButton` somewhere in the error message here so we can see what it tried to do.